src/seedu/addressbook/data/person/Address.java

```diff
@@ -8,24 +8,44 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
+    public static final String EXAMPLE = "123, some street,23 ,234";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses should have block, postal code, street and unit";
+    public static final String LIST_OF_4_CONSTRAINT_REGEX = "([a-zA-Z0-9#  -]+)(,[a-zA-Z0-9#  -]+){3}";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final int BLOCK_INDEX = 1;
+    public static final int POSTALCODE_INDEX = 2;
+    public static final int STREET_INDEX = 0;
+    public static final int UNIT_INDEX = 3;
 
     public final String value;
     private boolean isPrivate;
+    private Block block;
+    private PostalCode postalCode;
+    private Street street;
+    private Unit unit;
 
     /**
+     * @param address the address of the person.
+     *
      * Validates given address.
      *
-     * @throws IllegalValueException if given address string is invalid.
+     * @throws IllegalValueException if given address string is invalid or 4 valid Strings for Street,Block,Unit,Postal Code.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
-        this.isPrivate = isPrivate;
+
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
```

Comment:
How about using the smaller classes for their respective validation to increase precision?